package/gluon-core/files/lib/netifd/proto/gluon_wired.sh

```diff
@@ -28,10 +42,22 @@ proto_gluon_wired_setup() {
 
         local meshif="$config"
 
-        local vxlan="$(lua -e 'print(require("gluon.site").mesh.vxlan(true))')"
+        local transitive index vxpeer6addr is_tunnel
+        json_get_vars transitive index vxpeer6addr is_tunnel
+
+        ( proto_add_host_dependency "$config" '' "$ifname" )
 
-        local transitive index
-        json_get_vars transitive index
+        # default args
+        [ -z "$is_tunnel" ] && is_tunnel=0
+        [ -z "$vxpeer6addr" ] && vxpeer6addr='ff02::15c'
+
+        if [ "$is_tunnel" -eq 1 ] && is_layer3_device "$ifname"; then
```

Comment:
Can you explain what `is_tunnel` represents? I'm not sure I follow the reasoning here.

Reply:
It is possible that `vxlan` for *mesh_on_lan* and *mesh_on_wan* is disabled in the site/domain.

However for vpn we want to have vxlan if the interface is a layer 3 vpn and disabled if we have a layer 2 vpn. There is no correlation to the `vxlan` setting from site.

As I was not able to get the information whether an interface is a tunnel from somewhere else, I introduced the proto variable `is_tunnel`.

Reply:
I wonder if we can get read `IFF_TUN` and `IFF_TAP` from `/sys/class/net/$ifname/tun_flags`.

https://elixir.bootlin.com/linux/v4.14.213/source/include/uapi/linux/if_tun.h#L62

Reply:
This does not seem to exist for wireguard devices.

Reply:
No sensible way I see to reliably detect a tunnel interface in a consistent fashion.

Reply:
In libgluonutil we have this code: https://github.com/freifunk-gluon/gluon/blob/master/package/libgluonutil/src/libgluonutil.c#L156-L202

Reply:
Not necessary anymore.